### PR TITLE
Add new prefix: nasaosdr

### PR DESCRIPTION
**Pull request by nagutm**
PubMed: https://pubmed.ncbi.nlm.nih.gov/39558178/
Database: https://osdr.nasa.gov/bio/repo/search?q=&data_source=cgene,alsda&data_type=study

This pull request addresses the following issue: https://github.com/biopragmatics/bioregistry/issues/1293

This resource re-labels existing entries from NASA's GeneLab (`ngl`) resource with new identifier formats.

The main difference in identifier formats is the first few letters of the identifier. 'GLDS' for the old GeneLab and 'OSD' for the new 
Open Science Data Repository (OSDR). The numbers proceeded by the letters remain consistent across the two formats. 

Resolving an old identifier pattern with the old uri_format pattern from GeneLab automatically redirects you to the corresponding page on the OSDR.

### CODE REVIEW

**Comment by bgyori** [comment]
Upon review, it seems like `nasa.osdr` is an appropriate prefix for the overall resource but it actually seems to have multiple semantic spaces:
- studies: https://osdr.nasa.gov/bio/repo/data/studies/OSD-678
- payloads: https://osdr.nasa.gov/bio/repo/data/payloads/CARA
- missions: https://osdr.nasa.gov/bio/repo/data/missions/SpaceX-3

and potentially more. The current PR just curates studies. 

Some possible solutions:
- Introduce a third level to the prefix i.e. `nasa.osdr.study` and then optionally also curate similar prefixes for payloads, missions, etc.
- If for some reason prefixes with two dots in them are not a good idea, then we could call the base prefix `nasaosdr` and then have subprefixes like `nasaosdr.study`, etc.

**Comment by nagutm** [comment]
It seems there are 8 distinct semantic spaces associated with this resource. Of these 8 resources, 7 of them have actual identifiers associated with their entries, and the only one that does not is the `vehicle` subspace which reuses the name of the vehicle as its 'identifier' [example](https://osdr.nasa.gov/bio/repo/data/vehicles/Scout%20satellite) and so I have not curated this. I have highlighted the key points for each subspace below for further consideration of whether it is relevant to the Bioregistry.

- The `study` and `experiment` subspaces seem to be the most relevant with identifiers for studies and experiments related to space flight. They have consistent identifier formats for all of their entries and mirror the old `ngl` prefix.

- The  `hardware`, `subject` and `biospecimen` subspaces provide identifiers for very general entities. For example, the subject subspace has an entry for [wheat](https://osdr.nasa.gov/bio/repo/data/subjects/58), the biospecimen subspace has an entry for [rat liver](https://osdr.nasa.gov/bio/repo/data/biospecimens/7943) and the hardware subspace has an entry for [8 Foot Centrifuge](https://osdr.nasa.gov/bio/repo/data/hardware/2011)

- The `mission` and `payload` subspaces provide identifiers for the corresponding mission and payload associated with specific experiments. Their identifiers formats are quite random and curated with the most general regex pattern of `^[\\w\\W]+$` which might be considered too open-ended.

File: src/bioregistry/data/bioregistry.json [hunk 1]
```diff
@@ -74590,6 +74590,212 @@
...
+    "description": "Identifiers for missions associated with experiments that investigate biological and health responses of terrestrial life to spaceflight",
+    "example": "ARCBedrest",
+    "example_extras": [
+      "STS-46",
+      "Bion 11"
+    ],
+    "github_request_issue": 1298,
+    "homepage": "https://www.nasa.gov/osdr",
+    "name": "NASA Open Science Data Repository",
+    "pattern": "^[\\w\\W]+$",
```
Comment (bgyori) [inline src/bioregistry/data/bioregistry.json:74698]:
```suggestion
    "pattern": "^.+$",
```

**Review by bgyori — APPROVED** [review]
Loos good! I changed a couple of the regexps, made the names specific to each prefix, and fixed one uri_pattern, then added reviewer info.

**Pull request merged**